iOS App - an error has occurred

### Vaultwarden Support String

### Your environment (Generated via diagnostics page)
* Vaultwarden version: v1.32.1
* Web-vault version: v2024.6.2c
* OS/Arch: linux/x86_64
* Running within a container: true (Base: Debian)
* Environment settings overridden: true
* Uses a reverse proxy: true
* IP Header check: true (X-Real-IP)
* Internet access: true
* Internet access via a proxy: false
* DNS Check: true
* Browser/Server Time Check: true
* Server/NTP Time Check: true
* Domain Configuration Check: true
* HTTPS Check: true
* Database type: SQLite
* Database version: 3.46.0
* Clients used: 
* Reverse proxy and version: 
* Other relevant information: 

### Config (Generated via diagnostics page)
<details><summary>Show Running Config</summary>

**Environment settings which are overridden:** ADMIN_TOKEN


```json
{
  "_duo_akey": null,
  "_enable_duo": false,
  "_enable_email_2fa": true,
  "_enable_smtp": true,
  "_enable_yubico": true,
  "_icon_service_csp": "",
  "_icon_service_url": "",
  "_ip_header_enabled": true,
  "_max_note_size": 10000,
  "_smtp_img_src": "cid:",
  "admin_ratelimit_max_burst": 3,
  "admin_ratelimit_seconds": 300,
  "admin_session_lifetime": 20,
  "admin_token": "***",
  "allowed_iframe_ancestors": "",
  "attachments_folder": "data/attachments",
  "auth_request_purge_schedule": "30 * * * * *",
  "authenticator_disable_time_drift": false,
  "data_folder": "data",
  "database_conn_init": "",
  "database_max_conns": 10,
  "database_timeout": 30,
  "database_url": "***************",
  "db_connection_retries": 15,
  "disable_2fa_remember": false,
  "disable_admin_token": false,
  "disable_icon_download": false,
  "domain": "*****://**********************",
  "domain_origin": "*****://**********************",
  "domain_path": "",
  "domain_set": true,
  "duo_context_purge_schedule": "30 * * * * *",
  "duo_host": null,
  "duo_ikey": null,
  "duo_skey": null,
  "duo_use_iframe": false,
  "email_2fa_auto_fallback": false,
  "email_2fa_enforce_on_verified_invite": false,
  "email_attempts_limit": 3,
  "email_change_allowed": true,
  "email_expiration_time": 600,
  "email_token_size": 6,
  "emergency_access_allowed": true,
  "emergency_notification_reminder_schedule": "0 3 * * * *",
  "emergency_request_timeout_schedule": "0 7 * * * *",
  "enable_db_wal": true,
  "enable_websocket": true,
  "enforce_single_org_with_reset_pw_policy": false,
  "event_cleanup_schedule": "0 10 0 * * *",
  "events_days_retain": null,
  "experimental_client_feature_flags": "fido2-vault-credentials",
  "extended_logging": true,
  "helo_name": null,
  "hibp_api_key": null,
  "http_request_block_non_global_ips": true,
  "http_request_block_regex": null,
  "icon_blacklist_non_global_ips": true,
  "icon_blacklist_regex": null,
  "icon_cache_folder": "data/icon_cache",
  "icon_cache_negttl": 259200,
  "icon_cache_ttl": 2592000,
  "icon_download_timeout": 10,
  "icon_redirect_code": 302,
  "icon_service": "internal",
  "incomplete_2fa_schedule": "30 * * * * *",
  "incomplete_2fa_time_limit": 3,
  "increase_note_size_limit": false,
  "invitation_expiration_hours": 120,
  "invitation_org_name": "Vaultwarden",
  "invitations_allowed": false,
  "ip_header": "X-Real-IP",
  "job_poll_interval_ms": 30000,
  "log_file": null,
  "log_level": "info",
  "log_timestamp_format": "%Y-%m-%d %H:%M:%S.%3f",
  "login_ratelimit_max_burst": 10,
  "login_ratelimit_seconds": 60,
  "org_attachment_limit": null,
  "org_creation_users": "",
  "org_events_enabled": false,
  "org_groups_enabled": false,
  "password_hints_allowed": true,
  "password_iterations": 100000,
  "push_enabled": false,
  "push_identity_uri": "https://identity.bitwarden.com",
  "push_installation_id": "***",
  "push_installation_key": "***",
  "push_relay_uri": "https://push.bitwarden.com",
  "reload_templates": false,
  "require_device_email": false,
  "rsa_key_filename": "data/rsa_key",
  "send_purge_schedule": "0 5 * * * *",
  "sendmail_command": null,
  "sends_allowed": true,
  "sends_folder": "data/sends",
  "show_password_hint": false,
  "signups_allowed": false,
  "signups_domains_whitelist": "",
  "signups_verify": true,
  "signups_verify_resend_limit": 6,
  "signups_verify_resend_time": 3600,
  "smtp_accept_invalid_certs": false,
  "smtp_accept_invalid_hostnames": false,
  "smtp_auth_mechanism": "Login",
  "smtp_debug": false,
  "smtp_embed_images": true,
  "smtp_explicit_tls": null,
  "smtp_from": "**********************",
  "smtp_from_name": "Vaultwarden",
  "smtp_host": "*************",
  "smtp_password": "***",
  "smtp_port": 587,
  "smtp_security": "starttls",
  "smtp_ssl": null,
  "smtp_timeout": 15,
  "smtp_username": "**********************",
  "templates_folder": "data/templates",
  "tmp_folder": "data/tmp",
  "trash_auto_delete_days": null,
  "trash_purge_schedule": "0 5 0 * * *",
  "use_sendmail": false,
  "use_syslog": false,
  "user_attachment_limit": null,
  "user_send_limit": null,
  "web_vault_enabled": true,
  "web_vault_folder": "web-vault/",
  "yubico_client_id": null,
  "yubico_secret_key": null,
  "yubico_server": null
}
```
</details>


### Vaultwarden Build Version

v1.32.1

### Deployment method

Official Container Image

### Custom deployment method

_No response_

### Reverse Proxy

traefik:v2.9

### Host/Server Operating System

Linux

### Operating System Version

Ubuntu 22.04.5 LTS

### Clients

iOS

### Client Version

2024.9.2 (1106)

### Steps To Reproduce

1. Install the app from the App Store
2. Change to self hosted server
3. Enter email and password
4. Perform 2FA (WebAuthn)

### Expected Result

Entries are successfully synchronized

### Actual Result

The error message “An error has occurred” appears

### Logs

```text
vault  | [2024-10-10 13:35:58.256][start][INFO] Rocket has launched from http://0.0.0.0:80
vault  | [2024-10-10 13:39:28.527][request][INFO] GET /api/config
vault  | [2024-10-10 13:39:28.528][response][INFO] (config) GET /api/config => 200 OK
vault  | [2024-10-10 13:39:28.528][request][INFO] GET /api/config
vault  | [2024-10-10 13:39:28.529][response][INFO] (config) GET /api/config => 200 OK
vault  | [2024-10-10 13:39:34.825][request][INFO] GET /api/devices/knowndevice
vault  | [2024-10-10 13:39:34.827][response][INFO] (get_known_device) GET /api/devices/knowndevice => 200 OK
vault  | [2024-10-10 13:39:51.383][request][INFO] POST /identity/accounts/prelogin
vault  | [2024-10-10 13:39:51.385][response][INFO] (prelogin) POST /identity/accounts/prelogin => 200 OK
vault  | [2024-10-10 13:39:51.482][request][INFO] POST /identity/connect/token
vault  | [2024-10-10 13:39:51.555][error][ERROR] 2FA token not provided
vault  | [2024-10-10 13:39:51.555][response][INFO] (login) POST /identity/connect/token => 400 Bad Request
vault  | [2024-10-10 13:39:59.213][request][INFO] POST /identity/connect/token
vault  | [2024-10-10 13:39:59.553][vaultwarden::api::identity][INFO] User username@example.tld logged in successfully. IP: xxx.xxx.xxx.xxx
vault  | [2024-10-10 13:39:59.553][response][INFO] (login) POST /identity/connect/token => 200 OK
vault  | [2024-10-10 13:39:59.673][request][INFO] GET /api/config
vault  | [2024-10-10 13:39:59.673][response][INFO] (config) GET /api/config => 200 OK
vault  | [2024-10-10 13:39:59.872][request][INFO] GET /api/sync
vault  | [2024-10-10 13:39:59.896][response][INFO] (sync) GET /api/sync?<data..> => 200 OK
vault  | [2024-10-10 13:40:00.498][request][INFO] GET /api/sync
vault  | [2024-10-10 13:40:00.533][response][INFO] (sync) GET /api/sync?<data..> => 200 OK
```


### Screenshots or Videos

_(message says: An error occured)_
![screenshot-of-error-message](https://github.com/user-attachments/assets/463c08d2-37a7-4222-9d27-d0afecc37e6a)

### Additional Context

I have tried the following, but nothing has helped
- Version 1.32.0
- I read somewhere that it could be due to incorrect timestamps, so I updated every entry in the vault (edited and saved again directly without making any real changes)
- Changed KDF to 700.000
___
Login via web and desktop app works fine

--- a/src/db/models/cipher.rs
+++ b/src/db/models/cipher.rs
@@ -176,7 +176,27 @@ impl Cipher {
                     .inspect_err(|e| warn!("Error parsing fields {e:?} for {}", self.uuid))
                     .ok()
             })
-            .map(|d| d.into_iter().map(|d| d.data).collect())
+            .map(|d| {
+                d.into_iter()
+                    .map(|mut f| {
+                        // Check if the `type` key is a number, strings break some clients
+                        // The fallback type is the hidden type `1`. this should prevent accidental data disclosure
+                        // If not try to convert the string value to a number and fallback to `1`
+                        // If it is both not a number and not a string, fallback to `1`
+                        match f.data.get("type") {
+                            Some(t) if t.is_number() => {}
+                            Some(t) if t.is_string() => {
+                                let type_num = &t.as_str().unwrap_or("0").parse::<u8>().unwrap_or(1);
+                                f.data["type"] = json!(type_num);
+                            }
+                            _ => {
+                                f.data["type"] = json!(1);
+                            }
+                        }
+                        f.data
+                    })
+                    .collect()
+            })
             .unwrap_or_default();
 
         let password_history_json: Vec<_> = self
@@ -244,7 +264,7 @@ impl Cipher {
 
         // NOTE: This was marked as *Backwards Compatibility Code*, but as of January 2021 this is still being used by upstream
         // data_json should always contain the following keys with every atype
-        data_json["fields"] = Value::Array(fields_json.clone());
+        data_json["fields"] = json!([fields_json]);
         data_json["name"] = json!(self.name);
         data_json["notes"] = json!(self.notes);
         data_json["passwordHistory"] = Value::Array(password_history_json.clone());

